### PR TITLE
Add search

### LINKED ISSUE #7: Set up search
- [ ] index model fields (https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-extra-fields)
- [ ] provide search box and result view
- [x] decide on search backend

### PR DESCRIPTION
**Pull request by frcroth**
Closes #7 

The search page can be found at /search, we might want to discuss adding the search field to the base page, but that is a design question.

The filtering creates a bunch of warnings which is a little annoying, maybe they should be suppressed.

### CODE REVIEW

File: myhpi/core/models.py [hunk 1]
```diff
@@ -29,6 +30,11 @@ class BasePage(Page):
         FieldPanel("is_public", widget=forms.CheckboxInput),
         FieldPanel("visible_for", widget=forms.CheckboxSelectMultiple),
     ]
+    # FilterFields required for restricting search results
+    search_fields = Page.search_fields + [
+        index.FilterField("group_id"),
```
Comment (jeriox) [inline myhpi/core/models.py:35]: do we really want to search for the group id?

**Review by jeriox — COMMENTED** [review]
just the first thought, want to check it out and have a look in the next few days

File: myhpi/core/models.py (hunk 1 above)
Comment (frcroth) [reply myhpi/core/models.py:35]: No we don't. This field is required for permission checks (so we don't output search results the user can't visit)

Comment (jeriox) [reply myhpi/core/models.py:35]: I don't understand why this is working, as searching on filtered QuerySets should not be possible as of https://github.com/wagtail/wagtail/issues/6616. As ist works right now, I would be fine with accepting that even if it generates warnings

**Review by jeriox — APPROVED** [review]

**Comment by frcroth** [comment]
Merging now. Opens up #140 and #141

**Pull request merged**